tapeout/src/main/scala/transforms/RemoveUnusedModules.scala

```diff
@@ -39,21 +41,23 @@ class RemoveUnusedModulesPass extends Pass {
         case None => Set.empty[String]
       }
     }
-    val usedModuleSet = getUsedModules(modulesByName(c.main))
+    val usedModuleSet = getUsedModules(modulesByName(state.circuit.main))
 
-    val usedModuleSeq = c.modules.filter { usedModuleSet contains _.name }
+    val usedModuleSeq = state.circuit.modules.filter { usedModuleSet contains _.name }
+    val usedModuleNames = usedModuleSeq.map(_.name)
 
-    Circuit(c.info, usedModuleSeq, c.main)
-  }
-}
+    val renames = state.renames.getOrElse(RenameMap())
 
-class RemoveUnusedModules extends Transform with SeqTransformBased {
-  def inputForm = MidForm
-  def outputForm = MidForm
-  def transforms = Seq(new RemoveUnusedModulesPass)
+    //state.circuit.modules.filterNot { usedModuleSet contains _.name } foreach { x => renames.record(ModuleTarget(state.circuit.main, x.name), Seq()) }
```

Comment:
Note this should be the correct solution except for hack below.

Reply:
done